src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Intermediate/DocumentIntermediateNodeExtensions.cs

```diff
@@ -57,17 +59,56 @@ public static IReadOnlyList<IntermediateNodeReference> FindDirectiveReferences(t
         return visitor.Directives;
     }
 
-    public static IReadOnlyList<IntermediateNodeReference> FindDescendantReferences<TNode>(this DocumentIntermediateNode document)
+    public static ImmutableArray<IntermediateNodeReference> FindDescendantReferences<TNode>(this DocumentIntermediateNode document)
```

Comment:
>FindDescendantReferences [](http://example.com/codeflow?start=60&length=24)

I see calls to this that filter using a Where after the call. Would it make sense to have a version of the method that takes a filter predicate?